pkg/strategy/streaming_test.go

```diff
@@ -202,3 +236,145 @@ func TestAgentSidecarIsInjectedIntoQueryForStreaming(t *testing.T) {
+	assert.Contains(t, jaeger.Spec.Collector.Options.Map(), "kafka.producer.authentication")
+	assert.Contains(t, jaeger.Spec.Collector.Options.Map(), "kafka.producer.tls.key")
+	assert.Contains(t, jaeger.Spec.Collector.Options.Map(), "kafka.producer.tls.cert")
+	assert.Contains(t, jaeger.Spec.Collector.Options.Map(), "kafka.producer.tls.ca")
+	assert.Contains(t, jaeger.Spec.Collector.Options.Map(), "kafka.producer.topic")
+	assert.NotContains(t, jaeger.Spec.Collector.Options.Map(), "kafka.consumer.brokers")
+
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.brokers")
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.authentication")
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.tls.key")
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.tls.cert")
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.tls.ca")
+	assert.Contains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.consumer.topic")
+	assert.NotContains(t, jaeger.Spec.Ingester.Options.Map(), "kafka.producer.brokers")
+
+	assert.Len(t, jaeger.Spec.Volumes, 2)
+	assert.Len(t, jaeger.Spec.VolumeMounts, 2)
+}
+
+func TestReplaceVolume(t *testing.T) {
+	// prepare
+	existing := []corev1.Volume{
+		{
+			Name: "volume-a",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "secret-name-a",
+				},
+			},
+		}, {
+			Name: "volume-b",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "secret-name-b",
+				},
+			},
+		}, {
+			Name: "volume-c",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "secret-name-c",
+				},
+			},
+		},
+	}
+
+	new := []corev1.Volume{
+		{
+			Name: "volume-a",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "new-secret-name-a",
+				},
+			},
+		}, {
+			Name: "volume-b",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "new-secret-name-b",
+				},
+			},
+		},
+	}
+
+	// test
+	res := replaceVolume(existing, new...)
+
+	// verify
+	assert.Len(t, res, 3)
+
+	found := 0
+	for _, v := range res {
+		if v.Name == "volume-a" {
+			found = found + 1
+			assert.Equal(t, "new-secret-name-a", v.VolumeSource.Secret.SecretName)
+		}
+		if v.Name == "volume-b" {
+			found = found + 1
+			assert.Equal(t, "new-secret-name-b", v.VolumeSource.Secret.SecretName)
+		}
+	}
+	assert.Equal(t, 2, found)
+}
+
+func TestReplaceVolumeMount(t *testing.T) {
+	// prepare
+	existing := []corev1.VolumeMount{
+		{
+			Name:      "volume-a",
+			MountPath: "/var/path",
+		}, {
+			Name:      "volume-b",
+			MountPath: "/var/path-b",
+		}, {
+			Name:      "volume-c",
+			MountPath: "/var/path-c",
+		},
+	}
+
+	new := []corev1.VolumeMount{
+		{
+			Name:      "volume-a",
+			MountPath: "/user/path",
+		}, {
```

Comment:
Just out of curiosity, what is the reason for replacing two entries?

Reply:
The volume mounts are quite specific, with the contents being provided by the secret that is created by Strimzi based on the KafkUser resource. The names were quite generic, so, I prefixed the volumes/volume mounts with "kafkauser-". I opened a doc issue, to document that users should avoid creating a volume/volume mount prefixed with "kafkauser-", otherwise it might get overridden by the operator.